test/fixtures/es-module-loaders/hook-resolve-type.mjs

```diff
@@ -1,6 +1,29 @@
 let importedESM = 0;
 let importedCJS = 0;
-global.getModuleTypeStats = () => { return {importedESM, importedCJS} };
+
+export function globalPreload({ port }) {
+  port.on('message', (int32) => {
+    port.postMessage({ importedESM, importedCJS });
+    Atomics.store(int32, 0, 1);
+    Atomics.notify(int32, 0);
+  });
+  port.unref();
+  return `
+    const { receiveMessageOnPort } = getBuiltin('worker_threads');
+    global.getModuleTypeStats = async function getModuleTypeStats() {
+      const sab = new SharedArrayBuffer(4);
+      const int32 = new Int32Array(sab);
+      port.postMessage(int32);
+      // Artificial timeout to keep the event loop alive.
+      // https://bugs.chromium.org/p/v8/issues/detail?id=13238
+      // TODO(targos) Remove when V8 issue is resolved.
+      const timeout = setTimeout(() => {}, 1_000);
+      await Atomics.waitAsync(int32, 0, 0).value;
+      clearTimeout(timeout);
```

Comment:
nit: if we want to support cases where the response takes more than one second to come back (it's probably overkill for what this test is doing).
```suggestion
      let value;
      do {
        const timeout = setTimeout(() => {}, 1_000);
        value = await Atomics.waitAsync(int32, 0, 0, 999).value;
        clearTimeout(timeout);
      } while (value !== 'timed-out');
```

Reply:
I think it would indicate a problem if it took more than one second to come back.

Reply:
Like I said, for this test it's overkill, but if someone uses this test as a model for their own implementation, that could be problematic. Maybe instead we make the timeout throw to make it clear that we don't expect it to ever trigger:
```suggestion
      const timeout = setTimeout(() => { throw new Error('timeout') }, 1_000);
      await Atomics.waitAsync(int32, 0, 0).value;
      clearTimeout(timeout);
```

Reply:
@targos do you want to add my suggestion? It's not blocking, so feel free to reject it. In any case, this PR will need another CI run to be ready to land.